asciidoc-confluence-publisher-client/src/main/java/org/sahli/asciidoc/confluence/publisher/client/ConfluencePublisher.java

```diff
@@ -67,16 +72,37 @@ public void publish() {
         assertMandatoryParameter(isNotBlank(this.metadata.getSpaceKey()), "spaceKey");
         assertMandatoryParameter(isNotBlank(this.metadata.getAncestorId()), "ancestorId");
 
-        startPublishingUnderAncestorId(this.metadata.getPages(), this.metadata.getSpaceKey(), this.metadata.getAncestorId());
+        switch (this.publishingStrategy) {
+            case APPEND_TO_ANCESTOR:
+                startPublishingUnderAncestorId(this.metadata.getPages(), this.metadata.getSpaceKey(), this.metadata.getAncestorId());
+                break;
+            case REPLACE_ANCESTOR:
+                // verify that only a single root exists
+                if (this.metadata.getPages().size() > 1) {
+                    String rootPageTitles = this.metadata.getPages().stream().map(page -> "'" + page.getTitle() + "'").collect(joining(", "));
+                    throw new IllegalArgumentException("Multiple root pages detected: " + rootPageTitles + ", but '" + REPLACE_ANCESTOR + "' publishing strategy only supports one single root page");
+                }
+
+                if (this.metadata.getPages().size() > 0) {
+                    // replace ancestor title with single root page title
```

Comment:
Get rid of the comments (same for line 91).

Reply:
Fixed in aa52b05509206f322642a326c5ad05fb3c64825d